testing/include/gz/common/testing/TestPaths.hh

```diff
@@ -59,32 +61,30 @@ enum class GZ_COMMON_TESTING_VISIBLE BuildType
 ///
 /// It is intended that there is an implementation of this interface for
 /// each relevant buildsystem.
-class TestPaths
+class GZ_COMMON_TESTING_VISIBLE TestPaths
 {
   /// \brief Constructor
   /// \param[in] _projectSourcePath Path to the root of the project source
-  public: GZ_COMMON_TESTING_VISIBLE
-          explicit TestPaths(const std::string &_projectSourcePath =
-              kTestingProjectSourceDir);
+  public: explicit TestPaths(
+      const std::string &_projectSourcePath = kTestingProjectSourceDir);
 
   /// \brief Destructor
-  public: GZ_COMMON_TESTING_VISIBLE
-          virtual ~TestPaths() = 0;
+  public: virtual ~TestPaths() = 0;
```

Comment:
I noticed that there's a definition for this in `TestPaths.cc`. Should that be removed?

Reply:
Oh, yeah, we can remove that.

Reply:
Actually we may still need it: 

```
BazelTestPaths.cc.obj : error LNK2001: unresolved external symbol "public: virtual __cdecl gz::common::testing::TestPaths::~TestPaths(void)" (??1TestPaths@t
```

Reply:
Same on linux: /usr/bin/ld: 

```../lib/libgz-common6-testing.so.6.0.0: undefined reference to `gz::common::testing::TestPaths::~TestPaths()'```

Reply:
It's because it's a pure virtual destructor, so it needs a definition: https://en.cppreference.com/w/cpp/language/destructor

Reply:
TIL

Reply:
In this case, since there are other pure virtual functions, we could just make it a regular virtual destructor.

Reply:
097ee0b